tratto/src/main/java/star/tratto/util/FileUtils.java

```diff
@@ -0,0 +1,199 @@
+                    logger.error(errMsg);
+                }
+            }
+        } catch (IOException | FolderCreationFailedException | FileNotCreatedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void appendToFileExclusive(
+            String dirPath,
+            String fileName,
+            FileFormat fileFormat,
+            String projectName,
+            String content
+    ) {
+        String filePath = Paths.get(dirPath, projectName, fileName + fileFormat.getValue()).toString();
+        File file = new File(filePath);
+        boolean found = false;
+        if (file.exists()) {
+            try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    if (line.contains(content)) {
+                        found = true;
+                        break;
+                    }
+                }
+            } catch (IOException e) {
+                logger.error("Error reading file: " + e.getMessage());
+            }
+        }
+        if (!found) {
+            appendToFile(
+                    dirPath,
+                    fileName,
+                    fileFormat,
+                    projectName,
+                    content
+            );
+        }
+    }
+
+    /**
+     * The method creates a file {@link File} within a given directory.
+     *
+     * @param dirPath the path to the directory where the file must be saved
+     * @param fileName the name of the file where to write the content
+     * @param fileFormat the format of the file {@link FileFormat}.
+     *
+     * @return the file created {@link File}
+     * @throws IOException If the file cannot be created
+     */
+    public static File createFile(String dirPath, String fileName, FileFormat fileFormat) throws FolderCreationFailedException, IOException, FileNotCreatedException {
+        String filePath = Paths.get(dirPath, fileName + fileFormat.getValue()).toString();
+        File dir = new File(dirPath);
+        File file = new File(filePath);
+        // create directory.
+        if (!dir.exists()) {
+            boolean success = dir.mkdirs();
+            if (!success) {
+                String errMsg = String.format("Unable to create folder %s", dirPath);
+                logger.error(errMsg);
+                throw new FolderCreationFailedException();
+            }
+        }
+        // create file.
+        if (!file.exists()) {
+            boolean fileCreated = file.createNewFile();
+            if (!fileCreated) {
+                String errMsg = String.format("Unable to create file %s to path %s.", fileName, dirPath);
+                logger.error(errMsg);
+                throw new FileNotCreatedException();
+            }
+        }
+        return file;
+    }
+
+    /**
+     * The method gets the list of all the files that exist within a given directory.
```

Comment:
If this is recursive, please state that explicitly in the documentation.  The method name does not make it clear and could perhaps be improved.